### PR TITLE
Cognee integration

### PR DESCRIPTION
**Pull request by dexters1**
# Description

Integration of the cognee GraphRAG to LlamaIndex.
With added tests, README.md and usage example in README.md

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [x] I added new unit tests to cover this change
- [ ] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/poetry.lock
Comment (logan-markewich) [inline llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/poetry.lock:1]: no need to commit the lock file

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+
+class CogneeGraphRAG(GraphRAG):
+    def __init__(
+        self,
+        llm_api_key,
+        graph_db_provider,
+        vector_db_provider,
+        relational_db_provider,
+        db_name,
```
Comment (logan-markewich) [inline llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py:19]: None of these args are typed

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        )
+        cognee.config.data_root_directory(data_directory_path)
+        cognee_directory_path = str(
+            pathlib.Path(
+                os.path.join(pathlib.Path(__file__).parent, ".cognee_system/")
+            ).resolve()
+        )
+        cognee.config.system_root_directory(cognee_directory_path)
+
+    async def add(self, data, dataset_name: str):
```
Comment (logan-markewich) [inline llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py:43]: Can we add return type annotations, even its its `-> None` ?

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+from typing import List
+
+import cognee
+
+from llama_index.core import Document
+
+from .base import GraphRAG
+
+
+class CogneeGraphRAG(GraphRAG):
```
Comment (logan-markewich) [inline llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py:12]: Maybe a docstring is needed here

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/poetry.lock
Comment (dexters1) [reply llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/poetry.lock:1]: Removed lock file

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/tests/test_graph_rag_cognee.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import os
+import pandas as pd
+import asyncio
+
+import cognee
+from llama_index.core import Document
+from llama_index.graph_rag.cognee import CogneeGraphRAG
+
+
+async def test_graph_rag_cognee():
```
[the remark this answers is not in the document]
Comment (dexters1) [reply llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/tests/test_graph_rag_cognee.py:16]: Added marking as skipped if API key is missing

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py (hunk 2 above)
Comment (dexters1) [reply llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py]: Yes, it's added

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py (hunk 1 above)
Comment (dexters1) [reply llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py]: Added typing

**Comment by logan-markewich** [comment]
@dexters1 since the tests are skipped in CICD, actual coverage there is low. No worries though, understandable

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**